### PR TITLE
[WV][107226482] Record marker clicks

### PR DESCRIPTION
**Pull request by warrenv**
[Story](https://www.pivotaltracker.com/story/show/107226482)

Persist marker clicks to localStorage by saving the listing id.

The default is off and the app must set saveMarkerClick to true
in the markers options to enable it.
- [ ] javascript tests pass.

### CODE REVIEW

File: app/coffeescript/components/mixins/stored_markers.coffee [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  defineComponent
+) ->
+
+  storedMarkers = ->
+
+    @defaultAttrs
+      storageKey: 'viewedMapMarkers'
+
+    @load = ->
+      return {} unless Storage?
```
Comment (c0) [inline app/coffeescript/components/mixins/stored_markers.coffee]: Where does `Storage` come from?

Comment (warrenv) [inline app/coffeescript/components/mixins/stored_markers.coffee]: The browser, based on this http://stackoverflow.com/questions/11214404/how-to-detect-if-browser-supports-html5-local-storage

Comment (c0) [inline app/coffeescript/components/mixins/stored_markers.coffee]: :+1: 

What do you think about wrapping it in a function so we can clarify the intent? e.g.

``` coffee
localStorageSupported = ->
  Storage?
```

**Comment by pasha** [comment]
:+1:

**Pull request merged**